src/core/js/models/itemsComponentModel.js

```diff
@@ -0,0 +1,76 @@
+define([
+    'core/js/models/componentModel',
+    'core/js/models/itemModel'
+], function(ComponentModel, ItemModel) {
+
+    var ItemsComponentModel = ComponentModel.extend({
+
+        toJSON: function() {
+            var json = _.clone(this.attributes);
+
+            json._items = this.get('_items').toJSON();
+
+            return json;
+        },
+
+        init: function() {
+            this.setUpItems();
+
+            this.listenTo(this.get('_items'), {
+                'change:_isVisited': this.checkCompletionStatus
+            });
+        },
+
+        setUpItems: function() {
+            var items = this.get('_items');
+            var collection = new Backbone.Collection(null, { model: ItemModel });
+
+            items.forEach(function(item, index) {
+                item._index = index;
+                collection.add(item);
+            });
+
+            this.set('_items', collection);
+        },
+
+        getItem: function(index) {
+            return this.get('_items').findWhere({ _index: index });
+        },
+
+        getVisitedItems: function() {
+            return this.get('_items').where({ _isVisited: true });
+        },
+
+        getActiveItems: function() {
+            return this.get('_items').where({ _isActive: true });
+        },
+
+        getActiveItem: function() {
+            return this.get('_items').findWhere({ _isActive: true });
+        },
+
+        areAllItemsCompleted: function() {
+            return this.getVisitedItems().length === this.get('_items').length;
+        },
+
+        checkCompletionStatus: function() {
+            if (this.areAllItemsCompleted()) {
```

Comment:
Aha, I think defiite overkill / needless expansion for single-line if statements. Only an issue if dealing with multiple nested lines and/or actions after the conditional.